android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.kt

```diff
@@ -0,0 +1,511 @@
+    private const val DEFAULT_MIN_ALPHA_FOR_TRAVERSAL = 0f
+    private val tempPoint = PointF()
+    private val matrixTransformCoords = FloatArray(2)
+    private val inverseMatrix = Matrix()
+    private val tempCoords = FloatArray(2)
+    private val handlersComparator = Comparator<GestureHandler<*>?> { a, b ->
+      return@Comparator if (a.mIsActive && b.mIsActive || a.mIsAwaiting && b.mIsAwaiting) {
+        // both A and B are either active or awaiting activation, in which case we prefer one that
+        // has activated (or turned into "awaiting" state) earlier
+        Integer.signum(b.mActivationIndex - a.mActivationIndex)
+      } else if (a.mIsActive) {
+        -1 // only A is active
+      } else if (b.mIsActive) {
+        1 // only B is active
+      } else if (a.mIsAwaiting) {
+        -1 // only A is awaiting, B is inactive
+      } else if (b.mIsAwaiting) {
+        1 // only B is awaiting, A is inactive
+      } else {
+        0 // both A and B are inactive, stable order matters
+      }
+    }
+
+    private fun shouldHandlerlessViewBecomeTouchTarget(view: View, coords: FloatArray): Boolean {
+      // The following code is to match the iOS behavior where transparent parts of the views can
+      // pass touch events through them allowing sibling nodes to handle them.
+
+      // TODO: this is not an ideal solution as we only consider ViewGroups that has no background set
+      // TODO: ideally we should determine the pixel color under the given coordinates and return
+      // false if the color is transparent
+      val isLeafOrTransparent = view !is ViewGroup || view.getBackground() != null
+      return isLeafOrTransparent && isTransformedTouchPointInView(coords[0], coords[1], view)
+    }
+
+    private fun transformTouchPointToViewCoords(
+      x: Float,
+      y: Float,
+      parent: ViewGroup,
+      child: View,
+      outLocalPoint: PointF,
+    ) {
+      var localX = x + parent.scrollX - child.left
+      var localY = y + parent.scrollY - child.top
+      val matrix = child.matrix
+      if (!matrix.isIdentity) {
+        val localXY = matrixTransformCoords
+        localXY[0] = localX
+        localXY[1] = localY
+        matrix.invert(inverseMatrix)
+        inverseMatrix.mapPoints(localXY)
+        localX = localXY[0]
+        localY = localXY[1]
+      }
+      outLocalPoint[localX] = localY
+    }
+
+    private fun isTransformedTouchPointInView(x: Float, y: Float, child: View) =
+      x in 0f..child.width.toFloat() && y in 0f..child.height.toFloat()
+
+    private fun shouldHandlerWaitForOther(handler: GestureHandler<*>, other: GestureHandler<*>): Boolean {
+      return handler !== other && (handler.shouldWaitForHandlerFailure(other)
+        || other.shouldRequireToWaitForFailure(handler))
+    }
+
+    private fun canRunSimultaneously(a: GestureHandler<*>?, b: GestureHandler<*>) =
+      a === b || a!!.shouldRecognizeSimultaneously(b) || b.shouldRecognizeSimultaneously(a)
```

Comment:
The only usage of this method is in the one right below it and that one has non-nullable arguments.
```suggestion
    private fun canRunSimultaneously(a: GestureHandler<*>, b: GestureHandler<*>) =
      a === b || a.shouldRecognizeSimultaneously(b) || b.shouldRecognizeSimultaneously(a)
```

Reply:
Fixed in eaa3a0f.